File: tests/test-maven-generator/src/main/java/com/myapp1/model/SpringBeanRetriever.java
package com.myapp1.model;

import org.springframework.context.ApplicationContext;
import org.springframework.context.support.ClassPathXmlApplicationContext;

public class SpringBeanRetriever {

  private static SpringBeanRetriever instance = null;

  private ApplicationContext context;

  private SpringBeanRetriever() {
    this.context = new ClassPathXmlApplicationContext("spring-configuration.xml");
  }

  private static synchronized SpringBeanRetriever getInstance() {
    if (instance == null) {
      instance = new SpringBeanRetriever();
    }
    return instance;
  }

  public static <T> T getBean(final String name, final Class<T> c) {
    SpringBeanRetriever instance = SpringBeanRetriever.getInstance();
    return c.cast(instance.context.getBean(name));
  }

  @SuppressWarnings("unchecked")
  public static <T> T getBean(final String name) {
    SpringBeanRetriever instance = SpringBeanRetriever.getInstance();
    return (T) instance.context.getBean(name);
  }

}

File: attic/proof-of-concept/spring-mybatis-kotlin-poc/src/main/kotlin/shell/SpringBeanRetriever.kt
package shell

import org.springframework.context.ApplicationContext
import org.springframework.context.support.ClassPathXmlApplicationContext

class SpringBeanRetriever {

  private var context: ApplicationContext

  private constructor() {
    context = ClassPathXmlApplicationContext("spring-configuration.xml")
  }

  companion object { // static members here

    var instance3: SpringBeanRetriever? = null

    fun getInstance(): SpringBeanRetriever {
      if (instance3 == null) {
        instance3 = SpringBeanRetriever()
        return instance3!!
      } else {
        return instance3!!
      }
    }

    fun getBean(name: String): Any? {
      return SpringBeanRetriever.getInstance().context.getBean(name)
    }

  }

}
